### ISSUE TITLE
Is there have api for resourcegroup create?

### ISSUE THREAD

**Issue by xwzpp** [trinodb/trino#6814]
Is there have api for resourcegroup create?

**Comment by amitds1997** [comment]
No, there's no API to manage resource groups since the only current implementation of resource group manager is file-based.

**Comment by findepi** [comment]
There is also a second implementation which keeps state in a database, see `DbResourceGroupConfigurationManager` class.
I don't think it's currently documented though. Also, it's not as convenient to use as it could be, see #4191 for example improvement idea.

We should

**Comment by findepi** [comment]
There is also a second implementation which keeps state in a database, see `DbResourceGroupConfigurationManager` class.
I don't think it's currently documented though. Also, it's not as convenient to use as it could be, see #4191 for example improvement idea.

If the reason you're asking is because  you're interested in resource group reconfiguration without restart, we could enhance `FileSessionPropertyManagerConfig` with something like we have here:
https://github.com/trinodb/trino/blob/cea58426e0a8f4cfe330bc805f88d12ab2d3fc4e/lib[…]io/trino/plugin/base/security/FileBasedAccessControlConfig.java
(+ relevant implementation in `FileResourceGroupConfigurationManager` or somewhere near)

**Comment by findepi** [comment]
There is also a recent thread on this here: https://trinodb.slack.com/archives/CFLB9AMBN/p1613157788017400
